src/blockrelay/graphene_set.cpp

```diff
@@ -30,6 +31,7 @@ CGrapheneSet::CGrapheneSet(size_t _nReceiverUniverseItems,
 
     shorttxidk0 = _shorttxidk0;
     shorttxidk1 = _shorttxidk1;
+    ibltSalt = ibltEntropy % (uint8_t)std::max(1, (int)CIblt::MaxNHash());
```

Comment:
This is pedantic because MaxNHash is so small.  But if MaxNHash was > 65k, and certain ibltEntropy numbers were chosen, then ibltSalt could be > 65k, resulting in a salt*n_hashes > max int.  Shouldn't this eqn be:

ibltEntropy % (std::max<uint32_t>/std::max(1, (int)CIblt::MaxNHash());

A practical effect of using your eqn is that ibltSalt can really only vary from 0 to 8.  This isn't a lot of entropy for the effort.  I think we could have more.  But at the same time I don't like my eqn because there's no safety factor.  It would be better to ensure that throwing when salt*n_hashes > 0xffffffff will never happen in practice, except for nodes that give garbage values.

Reply:
Your comment made me take a second look at the logic in iblt.cpp that constructs hash seeds from the salt. When I originally wrote that code, I had a reason to ensure that different salts would produce disjoint sets of seeds. But that is no longer important, so I think this much simpler logic for seed `i` will suffice: `i + salt % (0xffffffff - n_hash)`, which ensures distinct seeds for each hash function. In light of that change, it is no longer necessary to perform any range checks on the salt. So I believe the line in question here becomes simply `ibltSalt = ibltEntropy`.